From 7f6f31d402bace2703c27f232fc67bb80afceabb Mon Sep 17 00:00:00 2001
From: Santiago Rossi <santiago_rho_psi@hotmail.com>
Date: Mon, 16 Apr 2012 21:30:59 -0300
Subject: [PATCH 29/40] 4.4.7 Cuenca de Sholazar: A los dioses les gustan las
 cosas brillantes

Support for quest "Gods like shiny things" from Oracles/Frenzyheart quest chain
Thanks to @oceanor for missing spawns
---
 ...2012_04_16_108_world_gods_like_shiny_things.sql |  115 ++++++++++++++++++++
 1 files changed, 115 insertions(+), 0 deletions(-)
 create mode 100644 sql/updates/world/reanemu/2012_04_16_108_world_gods_like_shiny_things.sql

diff --git a/sql/updates/world/reanemu/2012_04_16_108_world_gods_like_shiny_things.sql b/sql/updates/world/reanemu/2012_04_16_108_world_gods_like_shiny_things.sql
new file mode 100644
index 0000000..d7bfb3f
--- /dev/null
+++ b/sql/updates/world/reanemu/2012_04_16_108_world_gods_like_shiny_things.sql
@@ -0,0 +1,115 @@
+-- Missing spawns, Thanks to Oceanor
+DELETE FROM `creature` WHERE `id`=28362;
+INSERT INTO `creature` (`guid`, `id`, `map`, `spawnMask`, `phaseMask`, `modelid`, `equipment_id`, `position_x`, `position_y`, `position_z`, `orientation`, `spawntimesecs`, `spawndist`, `currentwaypoint`, `curhealth`, `curmana`,`MovementType`) VALUES
+(NULL, 28362, 571, 1, 1, 0, 0, 5551.06, 4750.33, -133.777, 1.81514, 180, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5607.36, 4355.13, -134.915, 4.45059, 180, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5726.51, 4370.49, -138.519, 3.10669, 180, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5606.67, 4545.54, -136.037, 4.95674, 180, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5693.35, 4629.68, -138.322, 4.92183, 180, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5432.09, 4315.3, -137.512, 4.67748, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5488.86, 4306.88, -130.141, 2.00713, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5479.12, 4363.98, -138.379, 4.17134, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5566.85, 4608.07, -140.005, 0.087266, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5733.78, 4598.79, -134.907, 0.802851, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5586.91, 4715.22, -132.39, 0.942478, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5789.39, 4640.83, -135.591, 5.53269, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5608.23, 4691.19, -136.457, 2.6529, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5494.54, 4597.56, -138.846, 3.05433, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5586.24, 4668.07, -134.59, 1.78024, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5743.72, 4645.34, -135.131, 3.07178, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5626.6, 4317.26, -133.309, 1.90241, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5663.73, 4370.93, -137.657, 4.72984, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5539.42, 4358.89, -136.533, 2.07694, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5586.07, 4504.85, -139.296, 0.820305, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5677.12, 4725.34, -135.566, 3.57792, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5507.06, 4662.85, -130.272, 2.44346, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5470.7, 4678.43, -132.203, 3.89208, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5704.71, 4763.64, -137.022, 5.0091, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5917.02, 4735.48, -137.81, 1.43117, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5544.33, 4436.61, -137.136, 4.24115, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5682.9, 4479.21, -135.683, 4.81711, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5579.44, 4388.16, -138.089, 3.80482, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5534.53, 4324.63, -131.82, 5.41052, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5906.66, 4612.75, -132.042, 2.60054, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5726.52, 4535.19, -138.999, 1.25664, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5525.25, 4436.55, -140.675, 1.95477, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5705.06, 4435.87, -141.029, 5.98648, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5847.39, 4599.29, -136.059, 5.53269, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5531.43, 4402.83, -135.817, 3.36848, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5700.94, 4711.59, -136.028, 4.04916, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5578.79, 4703.9, -132.344, 0.715585, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5865.93, 4563.6, -133.301, 3.52556, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5731.34, 4688.48, -134.257, 3.4383, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5594.8, 4733.19, -134.289, 0.10472, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5773.98, 4439.41, -137.158, 4.17134, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5863.84, 4759.62, -131.907, 5.16617, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5736.98, 4459.44, -134.957, 1.29154, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5806.82, 4481.87, -131.864, 4.60767, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5776.42, 4713.71, -132.204, 4.15388, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5794.37, 4567.62, -134.406, 4.59022, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5801.27, 4529.37, -134.282, 3.22886, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5760.65, 4690.13, -134.465, 5.58505, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5868.02, 4498.96, -133.209, 3.52556, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5652.97, 4494.59, -136.668, 2.60054, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5861.11, 4827.21, -131.635, 5.55015, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5786.58, 4751.95, -133.623, 0.767945, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5752.65, 4496.54, -132.448, 4.88692, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5738.6, 4740.34, -132.397, 1.36136, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5523.42, 4590.13, -133.389, 1.37881, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5634.23, 4725.39, -136.565, 4.67748, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5822.07, 4726.69, -132.864, 6.19592, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5616.78, 4507.51, -136.235, 2.79253, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5826.18, 4610.77, -134.161, 3.56047, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5523.1, 4547.95, -139.252, 0.034907, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5656.22, 4645.32, -136.027, 2.9147, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5807.02, 4767.91, -134.886, 3.56047, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5579.51, 4442.72, -139.873, 5.5676, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5656.49, 4679.36, -132.475, 0.855211, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5548.25, 4563.07, -136.596, 0.244346, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5751.76, 4767.77, -134.935, 3.97935, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5539.76, 4518.49, -134.643, 5.89921, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5538.02, 4635.59, -136.865, 1.0472, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5842.54, 4682.01, -134.929, 4.67748, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5768.06, 4605.94, -137.685, 3.87463, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5651.16, 4450.16, -137.938, 5.96903, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5699.09, 4671.95, -135.473, 5.98648, 300, 0, 0, 42, 0, 0),
+(NULL, 28362, 571, 1, 1, 0, 0, 5484.12, 4632.12, -136.877, 2.23402, 300, 0, 0, 42, 0, 0);
+
+-- Lafoo SAI
+SET @LAFOO := 28120;
+SET @DIRT := 28362;
+SET @BUG := 51443; -- Spell "Dig up Glinting Bug"
+SET @ARMOR := 51398; -- Spell "Dig up Glinting Armor"
+SET @GEM := 51400; -- Spell "Dig up Glowing Gem"
+SET @PLATTER := 51401; -- Spell "Dig up Polished Platter"
+SET @CRYSTAL := 51397; -- Spell "Dig up Shining Crystal"
+SET @HARE := 51441; -- Spell "Dig up Sparkling Hare"
+SET @TREASURE := 51402; -- Spell "Dig up Sparkling Treasure"
+UPDATE `creature_template` SET `AIName`='SmartAI' WHERE `entry` IN (@LAFOO,@DIRT);
+
+DELETE FROM `smart_scripts` WHERE `entryorguid` IN (@LAFOO,@DIRT) AND `source_type`=0;
+DELETE FROM `smart_scripts` WHERE `source_type`=9 AND `entryorguid` BETWEEN @LAFOO*100 AND @LAFOO*100+6;
+INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
+(@DIRT,0,0,1,10,0,100,1,1,10,0,0,45,0,1,0,0,0,0,19,@LAFOO,40,0,0,0,0,0,'Glinting dirt - LOS distance - Set data Lafoo'),
+(@DIRT,0,1,0,61,0,100,0,0,0,0,0,41,0,0,0,0,0,0,1,0,0,0,0,0,0,0,'Glinting dirt - LOS distance - Despawn'),
+(@LAFOO,0,0,0,1,0,100,1,0,0,0,0,29,3,0,0,0,0,0,23,0,0,0,0,0,0,0,'Lafoo - OOC - Follow summoner'),
+(@LAFOO,0,1,2,38,0,100,0,0,1,0,0,45,0,0,0,0,0,0,1,0,0,0,0,0,0,0,'Lafoo - Data set - Set data self'),
+(@LAFOO,0,2,3,61,0,100,1,0,0,0,0,1,0,0,0,0,0,0,1,0,0,0,0,0,0,0,'Lafoo - Follow completed - Set data Glowing Dirt'),
+(@LAFOO,0,3,0,61,0,100,0,0,0,0,0,88,@LAFOO*100,@LAFOO*100+6,0,0,0,0,1,0,0,0,0,0,0,0,'Lafoo - Follow completed - Run random script range'),
+(@LAFOO*100,9,0,0,0,0,100,0,0,0,0,0,11,@BUG,0,0,0,0,0,1,0,0,0,0,0,0,0,'Lafoo - Script - Cast Dig up Glinting Bug'),
+(@LAFOO*100+1,9,0,0,0,0,100,0,0,0,0,0,11,@ARMOR,0,0,0,0,0,1,0,0,0,0,0,0,0,'Lafoo - Script - Cast Dig up Glinting Armor'),
+(@LAFOO*100+2,9,0,0,0,0,100,0,0,0,0,0,11,@GEM,0,0,0,0,0,1,0,0,0,0,0,0,0,'Lafoo - Script - Cast Dig up Glowing Gem'),
+(@LAFOO*100+3,9,0,0,0,0,100,0,0,0,0,0,11,@PLATTER,0,0,0,0,0,1,0,0,0,0,0,0,0,'Lafoo - Script - Cast Dig up Polished Platter'),
+(@LAFOO*100+4,9,0,0,0,0,100,0,0,0,0,0,11,@CRYSTAL,0,0,0,0,0,1,0,0,0,0,0,0,0,'Lafoo - Script - Cast Dig up Shining Crystal'),
+(@LAFOO*100+5,9,0,0,0,0,100,0,0,0,0,0,11,@HARE,0,0,0,0,0,1,0,0,0,0,0,0,0,'Lafoo - Script - Cast Dig up Sparkling Hare'),
+(@LAFOO*100+6,9,0,0,0,0,100,0,0,0,0,0,11,@TREASURE,0,0,0,0,0,1,0,0,0,0,0,0,0,'Lafoo - Script - Cast Dig up Sparkling Treasure');
+
+
+DELETE FROM `creature_text` WHERE `entry`=@LAFOO;
+INSERT INTO `creature_text` (`entry`,`groupid`,`id`,`text`,`type`,`language`,`probability`,`emote`,`duration`,`sound`,`comment`) VALUES
+(@LAFOO,0,0,"Crunchy bugs so delicious. Want?",12,0,100,0,0,0,'Lafoo digging text'),
+(@LAFOO,0,1,"That weird. Someone took our bugs. You have maybe?",12,0,100,0,0,0,'Lafoo digging text'),
+(@LAFOO,0,2,"Maybe more shiny bugs! We like bugs!",12,0,100,0,0,0,'Lafoo digging text'),
+(@LAFOO,0,3,"Ooh! Shinies!",12,0,100,0,0,0,'Lafoo digging text'),
+(@LAFOO,0,4,"Treasure for Lafoo! Treasure for shrine!",12,0,100,0,0,0,'Lafoo digging text'),
+(@LAFOO,0,5,"Dig, dig, dig...",12,0,100,0,0,0,'Lafoo digging text');
\ No newline at end of file
-- 
1.7.8.msysgit.0

